api/verify_test.go

```diff
@@ -47,10 +49,10 @@ func (ts *VerifyTestSuite) SetupTest() {
 	require.NoError(ts.T(), ts.API.db.Create(u), "Error saving new test user")
 
 	// Create corresponding identity
-	i, err := ts.API.createNewIdentity(ts.API.db, u, "email", map[string]interface{}{
-		"sub":   u.ID.String(),
-		"email": u.GetEmail(),
-	})
+	i, err := ts.API.createNewIdentity(ts.API.db, u, "email", structs.Map(provider.Claims{
+		Subject: u.ID.String(),
+		Email:   u.GetEmail(),
+	}))
```

Comment:
I don't mind the `structs` library -- even if it is unmaintained -- but I feel like there's a simpler approach to this. All it does ends up as an input to this function.

Why don't we make the `identityData` here an official struct, something like:

```go
type IdentityData struct {
  Subject string `json:"sub"`
  Email string `json:"email"`
}
```

And use that instead? Shouldn't it be simpler that way?

Reply:
yeah sounds like a good idea, let me try it out

Reply:
Hmm actually, i think it might be better to not restrict the `identity_data` column to the `IdentityData` struct. For example, in the `IdTokenGrant` (https://github.com/supabase/gotrue/blob/master/api/token.go#L476), the claims used here can contain custom fields which we might not have included in the struct